miner/proposal_builder.go

```diff
@@ -259,7 +259,7 @@ func (pb *ProposalBuilder) createProposal(
 	}
 	p.Ballot.Signature = pb.signer.Sign(signing.BALLOT, p.Ballot.SignedBytes())
 	p.Signature = pb.signer.Sign(signing.BALLOT, p.SignedBytes())
-	p.SetSmesherID(pb.signer.NodeID())
+	p.SmesherID = pb.signer.NodeID()
```

Comment:
you need to move this up before signing the proposal.

Reply:
unfortunately, putting SmesherID inside Ballot meaning it's part of the proposal signature.

Reply:
why doesn't unittest catch this error?

Reply:
Fixed and updated unit tests to detect if the signature of a proposal cannot be verified